### ISSUE TITLE
Is there a way to add Metadata to a created Transaction

### ISSUE THREAD

**Issue by gavinharris-dev** [Hyperion-BT/Helios#16]
Cardano Transactions are able to contain additional data (auxiliary data) as the last part of the Transaction Block:

```
transaction =
  [ transaction_body
  , transaction_witness_set
  , bool
  , auxiliary_data / null
  ]
```

I cannot see any way to actually add this data into the current implementation of `Tx`. Is this possible (and something I have perhaps missed?) or is it a feature that has yet to be implemented?

Ref:
https://github.com/input-output-hk/cardano-ledger/blob/master/eras/babbage/test-suite/cddl-files/babbage.cddl

**Comment by christianschmitz** [comment]
Yet to be implemented, wasn't seen as a priority at the time (figuring out the precise data format isnt trivial)

**Comment by gavinharris-dev** [comment]
FMI: https://developers.cardano.org/docs/get-started/cardano-serialization-lib/transaction-metadata/

**Comment by gavinharris-dev** [comment]
Would you accept a simple setMetadata implementation that just allows a developer to provide an arbitrary Number [] for inclusion in a transaction? 
Can look at providing a Metadata builder class in the future!

**Comment by christianschmitz** [comment]
I was thinking of mapping javascript primitive types directly to meta-data

**Comment by christianschmitz** [comment]
Implemented as of v0.8.11
